src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -536,84 +536,97 @@ protected override async Task<HttpResponseMessage> SendAsync (HttpRequestMessage
+		public ClientCertificateOption ClientCertificateOptions { get; set; }
 
-		public ICredentials DefaultProxyCredentials {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L152
+		public ICredentials DefaultProxyCredentials { get; set; }
 
 		public int MaxAutomaticRedirections {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
+			get => int.MaxValue;
+			set {
+				// I believe it's possible to implement support for MaxAutomaticRedirections (it just has to be done)
+				if (value != int.MaxValue)
+					throw new ArgumentOutOfRangeException (nameof (value), value, "It's not possible to lower the max number of automatic redirections.");;
+			}
 		}
 
-		public int MaxConnectionsPerServer {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L154
+		public int MaxConnectionsPerServer { get; set; } = int.MaxValue;
 
-		public int MaxResponseHeadersLength {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L156
+		public int MaxResponseHeadersLength { get; set; } = 64; // Units in K (1024) bytes.
 
+		// We don't support PreAuthenticate, so always return false, and ignore any attempts to change it.
 		public bool PreAuthenticate {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
+			get => false;
+			set { }
 		}
 
-		public IDictionary<string, object> Properties {
-			get => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L167
+		public IDictionary<string, object> Properties { get { return null; } }
 
+		// We dont support any custom proxies, and don't let anybody wonder why their proxy isn't
+		// being used if they try to assign one (in any case we also return false from 'SupportsProxy').
 		public IWebProxy Proxy {
-			get => throw new PlatformNotSupportedException ();
+			get => null;
 			set => throw new PlatformNotSupportedException ();
```

Comment:
If you're going to throw on the setter, I'm not sure there's much point in returning null on the proxy?
Perhaps also consider using `[UnsupportedOSPlatform("ios")]` to warn people not to use it?